frontend/src/metabase/components/Schedule/constants.ts

```diff
@@ -2,81 +2,145 @@ import { c, t } from "ttag";
 
 export const minutes = times(60, n => ({
   label: n.toString(),
   value: n.toString(),
 }));
 
-const addZeroesToHour = (
-  hour: number,
-  { useZero = false }: { useZero: boolean },
-) => {
-  if (!useZero && hour === 0) {
-    hour = 12;
-  }
-  return c("This is a time like 12:00pm. {0} is the hour part of the time")
-    .t`${hour}:00`;
-};
-
 export const getHours = () => {
+  const localizedHours = [
+    c("A time on a 24-hour clock").t`0:00`,
+    c("A time").t`1:00`,
+    c("A time").t`2:00`,
+    c("A time").t`3:00`,
+    c("A time").t`4:00`,
+    c("A time").t`5:00`,
+    c("A time").t`6:00`,
+    c("A time").t`7:00`,
+    c("A time").t`8:00`,
+    c("A time").t`9:00`,
+    c("A time").t`10:00`,
+    c("A time").t`11:00`,
+    c("A time").t`12:00`,
+    c("A time on a 24-hour clock").t`13:00`,
+    c("A time on a 24-hour clock").t`14:00`,
+    c("A time on a 24-hour clock").t`15:00`,
+    c("A time on a 24-hour clock").t`16:00`,
+    c("A time on a 24-hour clock").t`17:00`,
+    c("A time on a 24-hour clock").t`18:00`,
+    c("A time on a 24-hour clock").t`19:00`,
+    c("A time on a 24-hour clock").t`20:00`,
+    c("A time on a 24-hour clock").t`21:00`,
+    c("A time on a 24-hour clock").t`22:00`,
+    c("A time on a 24-hour clock").t`23:00`,
+  ];
   const isClock24Hour = has24HourModeSetting();
   return times(isClock24Hour ? 24 : 12, n => ({
-    label: addZeroesToHour(n, { useZero: isClock24Hour }),
+    label: localizedHours[n],
     value: `${n}`,
   }));
 };
 
-export const optionNameTranslations = {
-  // The context is needed because 'hourly' can be an adjective ('hourly rate') or adverb ('update hourly'). Same with 'daily', 'weekly', and 'monthly'.
-  hourly: c("adverb").t`hourly`,
-  daily: c("adverb").t`daily`,
-  weekly: c("adverb").t`weekly`,
-  monthly: c("adverb").t`monthly`,
-};
-
 export type Weekday = {
   label: string;
   value: ScheduleDayType;
 };
 
-export const weekdays: Weekday[] = [
-  { label: t`Sunday`, value: "sun" },
-  { label: t`Monday`, value: "mon" },
-  { label: t`Tuesday`, value: "tue" },
-  { label: t`Wednesday`, value: "wed" },
-  { label: t`Thursday`, value: "thu" },
-  { label: t`Friday`, value: "fri" },
-  { label: t`Saturday`, value: "sat" },
-];
+/** These strings are created in a function, rather than in module scope, so that ttag is not called until the locale is set */
+export const getScheduleStrings = () => {
+  const scheduleOptionNames = {
+    // The context is needed because 'hourly' can be an adjective ('hourly rate') or adverb ('update hourly'). Same with 'daily', 'weekly', and 'monthly'.
+    hourly: c("adverb").t`hourly`,
+    daily: c("adverb").t`daily`,
+    weekly: c("adverb").t`weekly`,
+    monthly: c("adverb").t`monthly`,
+  };
+
+  const weekdays: Weekday[] = [
+    {
+      label: c(
+        "Occurs in phrases like 'Send weekly on Sunday'. Only capitalize if necessary",
+      ).t`Sunday`,
+      value: "sun",
```

Comment:
Some translators have used capitalized forms of "Sunday", "Monday", etc., even in languages where these are not standardly capitalized.